src/libraries/System.Private.CoreLib/src/System/Text/StringBuilder.cs

```diff
@@ -355,7 +355,7 @@ public override string ToString()
                     // Check that we will not overrun our boundaries.
                     if ((uint)(chunkLength + chunkOffset) > (uint)result.Length || (uint)chunkLength > (uint)sourceArray.Length)
                     {
-                        throw new ArgumentOutOfRangeException(nameof(chunkLength), SR.ArgumentOutOfRange_Index);
+                        throw new ArgumentOutOfRangeException(nameof(chunkLength), SR.ArgumentOutOfRange_IndexMustBeLessOrEqual);
```

Comment:
this is a wierd error because it means the StringBuilder is corrupt .. there's not really any index involved. Still your message isn't worse.